src/controllers/UserController.js

```diff
@@ -0,0 +1,37 @@
+const asyncHandler = require("express-async-handler");
+const User = require("../database/UserModel");
+
+// @desc   Adds new user if it doesnt exist and returns newly created user, otherwise return existing user from database
+// @route  PUT /users
+// @access Public
+const addOrUpdateUser = asyncHandler(async (req, res) => {
+  const { email, name } = req.body;
+  // Finds a user based off email and name and returns an object with only the id, email, name, recipes fields
+  let user = await User.findOne({ email: email, name: name }, "_id email name recipes").exec();
+  if (user) {
+    res.status(200).json({ 
+      data: user, 
+      message: "User already exists. Returning user information" 
+    });
+  } else {
+    user = await User.create({ email: email, name: name, recipes: [] });
+    if (user) {
+      res.status(200).json({
+        data: {
+          _id: user._id,
```

Comment:
this organization is not ideal. You are returning all the user's recipes when you just want to get the user's info (presumably an identifier that allows you to retrieve more information about the user for later).

right now you only have a recipes page, but in the future if/when you have other things, you'll be passing and processing a lot of information across the wire that are not needed for the page to function (e.g. sending recipes info for a schedule/calendar page)

Reply:
acknowledged and modified